components/style/servo/url.rs

```diff
@@ -126,3 +127,25 @@ impl ToCss for SpecifiedUrl {
         dest.write_str(")")
     }
 }
+
+impl ToComputedValue for SpecifiedUrl {
+    type ComputedValue = ComputedUrl;
+
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        match self.resolved {
+            Some(ref url) => ComputedUrl::Valid(url.clone()),
+            None => ComputedUrl::Invalid(Arc::new("about:invalid".to_string())),
```

Comment:
Why `about:invalid`, and not the specified one? Off-hand, that should be both cheaper and more correct.

What do other browsers do here?